### DISCUSSION TITLE
Connection through bastion?

### DISCUSSION THREAD

**Discussion by bmlab-sg** [nextflow-io/nf-sqldb#7] [Q&A]
Hi! 

Great plugin, really useful!
Small question, is that possible to connect to DB (which is isolated within VPC with no public access) through bastion host? 
Thanks!

**Comment by pditommaso** [comment]
It should be possible using a SSH tunnelling, for example https://aws.amazon.com/premiumsupport/knowledge-center/rds-connect-using-bastion-host-linux/

**Comment by bmlab-sg** [comment]
Thanks for the answer. Yes, the connection itself is working fine. I can ssh and connect to the DB. 
But I'm not sure how to execute it in the Nextflow context. Should I ssh each time prior to running NF (not very practical)? Can I embed it somehow into NF config/process?

**Comment by bmlab-sg** [comment]
Cool, its working fine!
For future reference if anyone is looking for it, I used

```
sshpass -p <PASSWORD> ssh -N -f <USERNAME>@<PROXY> -L 3307:<DB_HOST>:3306
```

and the config:

```
sql {
    db {
        '<DB_NAME>' {
              url = 'jdbc:mysql://127.0.0.1:3307/<DB_NAME>'
              user = '<DB_USER>'
              password = '<DB_PASS>'
            }
    }
}
```

Cheers!
